app/services/add_ephemera_to_collection.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+class AddEphemeraToCollection
+  attr_reader :project_id, :collection_id, :change_set_persister, :logger, :query_service
+  def initialize(project_id:, collection_id:, change_set_persister:, logger: Valkyrie.logger)
+    @project_id = project_id
+    @collection_id = collection_id
+    @change_set_persister = change_set_persister
+    @logger = logger
+    @query_service = Valkyrie.config.metadata_adapter.query_service
+  end
+
+  def process_box(box)
+    logger.info("Processing a box: #{box.decorate.title}")
+    box.decorate.members.each do |folder|
+      process_folder(folder)
+    end
+  end
+
+  def process_folder(folder)
+    logger.info("Processing a folder: #{folder.decorate.title.first}")
+    col_ids = folder.member_of_collection_ids + [collection_id]
+    change_set = ChangeSet.for(folder)
+    change_set_persister.save(change_set: change_set) if change_set.validate(member_of_collection_ids: col_ids)
+  end
+
+  def perform_now
```

Comment:
Can you use a different method name to avoid confusion with ActiveJob perform/perform_now? Having a pretty generic method like `execute` or whatever is fine, if there's nothing more semantically appropriate.

Reply:
Good idea